### DISCUSSION TITLE
Toggle live notifications from context menu

### DISCUSSION THREAD

**Discussion by x9136** [Chatterino/chatterino2#4831] [Ideas]
Some people watch streamers when there is an interesting game or category on the channel. For example, if you are only interested in Dota 2, and the streamer has stopped playing it, then it's better to turn off the notifications, so that you don't get distracted. When regular Dota 2 launches again, then turn on the notifications. 

Another case is when streamers make an announcement of a walkthrough. Let's say the notifications were turned on. Then, if the announcement interests you, then leave it as it is, and if not, then immediately turn it off and go to see what's going on there when you finish your business.

It's also more convenient than writing the streamer's name in full every time.

Of course, if you develop this idea, you will get [notifications about changing categories](https://github.com/Chatterino/chatterino2/discussions/4291), but it seems to be more complicated than the toggle in the context menu.

**Comment by Nerixyz** [comment]
You can already toggle live notifications from the split header:

![firefox_2023-09-19_17-05-55](https://github.com/Chatterino/chatterino2/assets/19953266/0bf43ba3-0d70-4cf9-b8bc-8a6077cf7cfc)

If that's not what you mean, could you go in more detail about the feature?

**Comment by x9136** [comment]
@Nerixyz It's too high (because the cursor is almost always near the input field) and you have to open a chat room to do it. I meant here: 
<img width="399" alt="Screenshot 2023-09-19 at 17 32 21" src="https://github.com/Chatterino/chatterino2/assets/60105060/beae32cc-8921-4a3c-9d3f-89cf94c3b5d5">

**Comment by Mm2PL** [comment]
I don't understand why the split header being high up on the screen is a problem.

You can always also add/remove channels to the list in the settings.
![image](https://github.com/Chatterino/chatterino2/assets/25011746/744a1cff-dc9f-40af-8616-ee60c9f59d0a)

###### You won't see the plugins tab if you've not compiled Chatterino with the right flags

**Comment by x9136** [comment]
> I don't understand why the split header being high up on the screen is a problem.

It's not as convenient as the bottom, because it takes longer to move there. There is modality, because you have to open each channel separately to toggle the notification. In the settings, you also have to write the whole nickname.

The fact that it's there is great, but it's not as convenient to use as it could be: without modality, next to the most frequent cursor place, without having to write nicknames in full every time.

I hope you are familiar with the problems of inconvenient interfaces.

**Reply by pajlada** [reply]
So the feature you're requesting is: When right-clicking a tab, I would like an option to mass apply a setting (in this case "Notify when live") to all channels open in the tab?

**Reply by x9136** [reply]
@pajlada No, I'm talking about being able to toggle the notification from the context menu of tabs, regardless of what channel you're in.

For example, there are 3 channels: channel 1, channel 2 and channel 3. Being in channel 1, I can call the context menu of any channel tab and switch the notification.

Another solution to the same problem is to make live notification when the right category appears on the stream. The problem I'm proposing to solve here is not just where the button is. **I think it's important to manage notifications to get them only when something you're interested in happens on the stream.** Having a button at your fingertips helps partially with that.

**Reply by Felanbird** [reply]
> Another solution to the same problem is to make live notification when the right category appears on the stream.

So this discussion is just a duplicate of #4291 - but you felt it necessary to throw in randomly moving a button to make it look less unnecessary?

I don't believe you need to add/remove channels from live notifications SO often that the current button is an inconvenience.

**Reply by Nerixyz** [reply]
> For example, there are 3 channels: channel 1, channel 2 and channel 3. Being in channel 1, I can call the context menu of any channel tab and switch the notification.

From this description, it seems really hard to know which channel you want to access. Say you have channels `a`, `b`, and `c` open and want to mute notifications in `c`. When you right-click on tab `b`, how would Chatterino know you want to access `c`? Displaying all channels won't work (as you can have a lot of channels open). Furthermore, keep in mind that one tab can contain multiple channels.

This might be a better fit for the quick-switcher (<kbd>CTRL</kbd>+<kbd>K</kbd>). It could be extended to have something like "mute channel" item where you can choose the channel.

**Comment by x9136** [comment]
@Felanbird

> but you felt it necessary to throw in randomly moving a button to make it look less unnecessary?

I was looking for a more convenient solution to the problem I mentioned above. That is, if it's easy enough to turn on and off the notifications of different channels, it's easy to customize when you will be disturbed and for what reason.

But, since this is a compromise and not a complete solution to the problem, I mention notifications for categories.

> I don't believe you need to add/remove channels from live notifications SO often that the current button is an inconvenience.

This is often enough for me to bother you and waste time discussing it, since from the outside it seems easy to add the option.

**Comment by x9136** [comment]
@Nerixyz

<img width="386" alt="Screenshot 2023-09-19 at 22 31 45" src="https://github.com/Chatterino/chatterino2/assets/60105060/b84863bb-acd7-4ec0-a4f2-69be0a9928b4">

Let's say these are the channels and it's clear where I am. If I remove the modality, I can right-click on any tab and toggle the notification of any channel, while staying where I am.

> Displaying all channels won't work (as you can have a lot of channels open). Furthermore, keep in mind that one tab can contain multiple channels.

Fair. I can agree that in these cases it will not be obvious what will happen.

> This might be a better fit for the quick-switcher (CTRL+K). It could be extended to have something like "mute channel" item where you can choose the channel.

You'll have to type instead of toggle, so it's best not to waste development time.

**Reply by Nerixyz** [reply]
One option I thought about was to add a button to disable notifications for a channel right from the live notification (with some confirmation).

**Reply by x9136** [reply]
I thought about this because it’s strange to delete, and the toggle is only in the context menu.

**Comment by x9136** [comment]
Thank you. The proposal is significantly flawed and of little help, so I suggest you don't waste your time.
